### ISSUE TITLE
第一个训练指令已经是使用ESOD了吗，评估结果有gap

### ISSUE THREAD

**Issue by DongRos** [alibaba/esod#11]
如题，训练指令为DATASET=visdrone MODEL=yolov5m GPUS=2 BATCH_SIZE=8 IMAGE_SIZE=1536 EPOCHS=50 bash ./scripts/train.sh
评估指令为python test.py --data data/visdrone.yaml --weights runs/train/exp2/weights/best.pt --img-size 1536 --device 0,1
只评估了visdrone数据集，评估输出结果如下图

![Image](https://github.com/user-attachments/assets/c3e71959-5967-49e9-8e8b-18b768486156)

显示map只有34.5，与论文的36.0有点差距，是我哪里设置的问题吗，还麻烦作者帮忙看看，感谢！

**Comment by DongRos** [comment]
我更改epoch为200重新训练后，在验证集上map提升到36.3，但是使用测试集进行评估后map降到了27，如图

![Image](https://github.com/user-attachments/assets/ed70c238-a9c3-4833-ab9b-d0e0d4c4dfbe)
是什么原因呢？这是使用了ESOD框架后的模型？？？

**Comment by DongRos** [comment]
使用作者提供的预训练模型在验证集评估  map33.1
​​

![Image](https://github.com/user-attachments/assets/ae171089-1c2e-4830-9953-4aa7acc270cd)
在测试集评估 map25.3

![Image](https://github.com/user-attachments/assets/f11e9cee-2de0-4e5f-8eb1-aba8b67d251a)

​​

**Comment by tso2381637** [comment]
Hi, I got the same result as yours. Have you figure out what's going on?

**Comment by yuhua666** [comment]
I also got the same result as yours.

**Comment by judycpChen** [comment]
Maybe you can try passing in some other parameters when running test.py, such as --use-gt. When I run the original inference command given by the author directly, the accuracy value on the visdrone validation set is 34.6, but when I pass in the --use-gt parameter, the accuracy value is 35.3.

<img width="1680" height="435" alt="Image" src="https://github.com/user-attachments/assets/4aaba226-f0c9-430b-9774-ef322888ba9c" />

<img width="1686" height="454" alt="Image" src="https://github.com/user-attachments/assets/e97fcbad-e3c3-44e0-8a84-b101f07afc01" />

**Comment by andy690** [comment]
> Maybe you can try passing in some other parameters when running test.py, such as --use-gt. When I run the original inference command given by the author directly, the accuracy value on the visdrone validation set is 34.6, but when I pass in the --use-gt parameter, the accuracy value is 35.3.
> 
> <img alt="Image" width="1680" height="435" src="https://private-user-images.githubusercontent.com/151814406/497384648-4aaba226-f0c9-430b-9774-ef322888ba9c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjIzMDg3NzgsIm5iZiI6MTc2MjMwODQ3OCwicGF0aCI6Ii8xNTE4MTQ0MDYvNDk3Mzg0NjQ4LTRhYWJhMjI2LWYwYzktNDMwYi05Nzc0LWVmMzIyODg4YmE5Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMTA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTEwNVQwMjA3NThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00NWI2ZGNmYzBhOWQyMGM5OWUwMWViODE5YmU1M2ZlZWU4NTRlNDNlNDdmYzNjMTYyMGUyZGE0ZjhmMTRlNmViJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.kvHh9d1uC_vqBetgQ5ukYfKaM-EHZ-HINJ4nvCljGMY"> <img alt="Image" width="1686" height="454" src="https://private-user-images.githubusercontent.com/151814406/497384664-e97fcbad-e3c3-44e0-8a84-b101f07afc01.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjIzMDg3NzgsIm5iZiI6MTc2MjMwODQ3OCwicGF0aCI6Ii8xNTE4MTQ0MDYvNDk3Mzg0NjY0LWU5N2ZjYmFkLWUzYzMtNDRlMC04YTg0LWIxMDFmMDdhZmMwMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMTA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTEwNVQwMjA3NThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05ZDAwZjcxYmJlYzE5YThjNDIxODA3MTMwNjYzMDY0OTQ4MzQ5MmQxOWNhN2M3YmQwNjNjOWI2ODQ3ZWRlMDM2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Y4l0Df2tgBXj6Lg9sF91AQLbMFQp7JaIUBCq05P5lTM">

这样会造成测试集的数据泄露，肯定是不能拿来当指标的

**Comment by andy690** [comment]
作者的实验设置中，测试集数量显示为3190，而我们用的测试集数量都为1610，为什么会有这样的差异？
